### ISSUE TITLE
looper should not require library, organism

### ISSUE THREAD

**Issue by nsheff** [epigen/looper#7]
right now we require a sample sheet to have columns called 'library' and 'organism', and the project config must have a 'genomes' and 'transcriptomes' section defining the genome for strings in 'organism'

however, with the new `derived_column` feature, looper becomes much more universal; it can also be used, for example, to run pipelines that combine 2, or 3, (or n) inputs.

for example, each row can be a comparison, (rather than a sample), and the yaml constructed would be a comparison representation rather than a sample. but if your pipeline expects a comparison rather than a sample, then it's no problem!

since a comparison has no element of "genome" necessarily, or "library", these are unnecessary columns.

in short, the require stuff above makes looper less flexible than it can/should be.

**Comment by afrendeiro** [comment]
Okay, implemented. Neither is required at all. 
Samples with an "organism" attribute will get "genome" and "transcriptome" attributes *if these mappings are specified*.
Looper will *create* `Sample` objects for samples without a "library" attribute, but looper will not be able to pair these with any pipeline and therefore skip submission with a warning.

**Comment by nsheff** [comment]
Ok, thanks. I will give it a try.
